app/lib/dartdoc/backend.dart

```diff
@@ -320,3 +320,58 @@ class DartdocBackend {
     await deletePool.close();
   }
 }
+
+class ArchiveTracker {
+  final _pvMap = <String, _TrackedEntry>{};
+  final _sampleLines = <String>[];
+
+  ArchiveTracker() {
+    _logger.info('Tracking: init.');
+    new Timer.periodic(const Duration(minutes: 10), (_) => _cleanup());
+    new Timer.periodic(const Duration(minutes: 45), (_) => _logSample());
+  }
+
+  List<String> get samples => _sampleLines;
+
+  void track(DartdocEntry entry) {
+    final key = '${entry.packageName}/${entry.packageVersion}';
+    final tracked = _pvMap.putIfAbsent(key, () => new _TrackedEntry());
+    tracked.archiveSize = entry.archiveSize ?? 0;
+    tracked.totalSize = entry.totalSize ?? 0;
+    tracked.lastAccess = new DateTime.now();
+  }
+
+  void _cleanup() {
+    final now = new DateTime.now();
+    _pvMap.removeWhere((key, value) {
+      final age = value.lastAccess.difference(now).abs();
+      return age.inMinutes > 90;
+    });
+  }
+
+  void _logSample() {
+    // roughly one and half days of samples
+    if (_sampleLines.length > 48) {
+      _sampleLines.removeAt(0);
+    }
+    final now = new DateTime.now();
+    final count = _pvMap.length;
+
+    final totalMB = _pvMap.values.fold(0, (sum, tracked) {
+      return sum + tracked.totalSize ~/ _mb;
+    });
+    final archiveMB = _pvMap.values.fold(0, (sum, tracked) {
+      return sum + tracked.archiveSize ~/ _mb;
+    });
+    _sampleLines.add(
+        '${now.toIso8601String()} count: $count, total: $totalMB MB, archive: $archiveMB MB');
```

Comment:
It doesn't say which tracker it is.

Suggestion: I suggest having a utility function that can pretty print an arbitrary byte quantity with the right unit. That will handle if the cache is very large of very small. I did that for another server side dart project and I like it.

Reply:
Now that you mention it, I shall have it somewhere too. Not sure if it is that important here: for the storage anything less than 100 MB is essentially the same. May be useful for the compressed archive length, but based on the staging datasets, MB range still gives us useful insights (e.g. total archive size is ~7MB for 120 packages -> ~58K per package on average).